File: app/src/main/java/com/itba/runningMate/repository/run/RunRepository.java
package com.itba.runningMate.repository.run;

import com.itba.runningMate.domain.Run;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Single;

public interface RunRepository {

    Flowable<List<Run>> getRun();

    Flowable<List<Run>> getRunLazy();

    Single<Run> getRun(final long uid);

    Single<Run> getRunMetrics(final long uid);

    Single<Long> insertRun(Run run);

    Single<Double> getTotalDistance();

    Single<Long> getMaxTime();

    Single<Double> getMaxKcal();

    Single<Double> getMaxSpeed();

    void deleteRun(Run run);

    void deleteRun(long runId);

    Completable updateTitle(long runId, String title);
}

File: app/src/main/java/com/itba/runningMate/repository/run/RunRepository.kt
package com.itba.runningMate.repository.run

import com.itba.runningMate.domain.Run
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Single

interface RunRepository {

    fun getRun(): Flowable<List<Run>>

    fun getRunLazy(): Flowable<List<Run>>

    fun getRun(uid: Long): Single<Run>

    fun getRunMetrics(uid: Long): Single<Run>

    fun insertRun(run: Run): Single<Long>

    fun getTotalDistance(): Single<Double>

    fun getMaxTime(): Single<Long>

    fun getMaxKcal(): Single<Double>

    fun getMaxSpeed(): Single<Double>

    fun deleteRun(run: Run): Completable

    fun deleteRun(runId: Long): Completable

    fun updateTitle(runId: Long, title: String): Completable

}